core/src/main/web/app/helpers/core.js

```diff
@@ -294,6 +294,7 @@
           preferencename: 'theme',
           preferencevalue: theme
         });
+        $rootScope.$broadcast('beaker.theme.set', theme);
```

Comment:
Use $rootScope.$emit instead of $broadcast. $emit will only trigger the listener on the $rootScope, while $broadcast will go through all of the child scopes. Since the listener is on rootScope, $emit is the correct option.

http://toddmotto.com/all-about-angulars-emit-broadcast-on-publish-subscribing/